### DISCUSSION TITLE
Optimizing Data Collection for Person Detection in RGB and Thermal Modalities

### DISCUSSION THREAD

**Discussion by nileshmetakota-17** [ultralytics/ultralytics#22443] [Solutions] [answered]
Hello Ultralytics Team, 

I’m currently focusing on the person class for detection and tracking tasks, using RGB (1920×1080) data captured during the day and thermal (640×480) data captured during both day and night. I’m not training all classes simultaneously — just optimizing for person detection.

Could you please share recommendations on the data collection strategy for this setup, especially when capturing thermal data in different palette modes (e.g., white hot, black hot, ironbow, etc.)?

Specifically, I’d like your input on:

Whether to collect still images or video recordings for training data.

If videos are preferred, what frame extraction rate (FPS) would yield optimal results.

The types of scenes and conditions to include — such as variations in lighting, background, distance, pose, and motion — to achieve YOLO-level detection accuracy.

My goal is to build a high-quality, well-generalized dataset for person detection across both RGB and thermal modalities. Any advice on the type, quality, quantity, or diversity of data that would strengthen the dataset would be greatly appreciated.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @nileshmetakota-17, thanks for starting this thoughtful discussion on optimizing person detection across RGB and thermal modalities 🚀. This is an automated response — an Ultralytics engineer will review and assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible so we can offer precise guidance. For your RGB + thermal person-detection setup, it would be especially helpful to include:
-Short samples of your images or brief video clips from both RGB (1920×1080) and thermal (640×480), including different palette modes (e.g., white hot, black hot, ironbow)
-Example annotations (format, tool used) and your class list (confirm single class: person)
-Train/val/test split details and total image/frame counts per modality
-The model and version you’re using (e.g., YOLO11n/s/m) and any pretrained weights
-Training config: image size, epochs, batch size, optimizer, augmentations, and any thermal-specific preprocessing
-Training logs and metrics (precision/recall/mAP, losses) plus a few validation predictions
-Your compute environment (CPU/GPU, CUDA/CUDNN, PyTorch, ultralytics version) and how you’re running training (Python/CLI)
-Intended deployment scenario (day/night ranges, distances, motion/pose variety) to align recommendations with your target use case

Also verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). Sharing these details will help us tailor guidance on data collection and curation for robust, well-generalized performance across modalities 💡

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply, marked as answer]
Thanks for the clear scope—focusing on a single `person` class across RGB + thermal is a solid plan. Prefer short videos per scene and extract frames offline to reduce redundancy; sample at a variable rate based on motion (low motion ≈ 1–3 FPS, high motion up to ≈ 5–10 FPS), and sprinkle in stills for rare edge cases. Keep thermal in a single canonical format if possible (ideally raw/grayscale normalized consistently); if you must support multiple palettes at inference, either train with balanced examples from each palette or convert palettes to a common grayscale during preprocessing to avoid domain drift. Aim for diversity across day/night, backgrounds, camera heights/angles, distances (near to small targets), occlusions, poses, and motion, plus some “negative” scenes with no people. You can mix modalities in one run by listing both roots in your `data.yaml` (keep counts roughly balanced):

```yaml
# person_rgb_thermal.yaml
names: [person]
train:
  - /data/rgb/images/train
  - /data/thermal/images/train
val:
  - /data/rgb/images/val
  - /data/thermal/images/val
```

```bash
yolo detect train data=person_rgb_thermal.yaml model=yolo11s.pt imgsz=640 epochs=100 batch=16 multi_scale=True
```

More curation tips (bias, splits, annotation consistency) are outlined in the Ultralytics guide on [Data Collection and Annotation Strategies](https://docs.ultralytics.com/guides/data-collection-and-annotation/); for a quick primer on modality trade-offs, see the overview of thermal vs. RGB in [Computer Vision Cameras: Types & Applications](https://www.ultralytics.com/blog/computer-vision-cameras-and-their-applications).
